docs/collator/SetupAndRun/keys.md

```diff
@@ -8,131 +8,179 @@ hide_title: false
 
 :::note
-both of the following methods (insert, rotate) use an unsafe rpc call to set the node session key. you must stop the service if it is running, then run the node with the `--rpc-methods=unsafe` parameter setting in order for the calls to succeed. don't forget to change the setting back to `safe` afterwards as a node that allows unsafe rpc calls and has an exposed rpc port can easily have its session keys changed by anyone, reulting in transaction fees being paid out somewhere other than where they are intended.
+While the Nimbus session key is present in the node since v.3.2.1, it will only become active with v3.3.0. Before this upgrade, the Aura session key below is used.
+:::
+
+In addition, your node might have two additional keys in storage:
+- **Aura Session Key**: Used in the node to decide on the authoring node *before* v3.3.0
+- **VRF Session Key**: A placeholder key for a future switch to a verifiably-random block producer selection scheme
+
+:::warning
+Both of the following methods (insert, rotate) use an **unsafe** RPC call to set the node session key.
+In order to use them, you must stop the node if running and start it with a `--rpc-methods=unsafe` CLI parameter.
+This mode of operation is **unsafe** when exposed to the public internet, restart the node without the CLI parameter when done.
 :::
 
 <Tabs groupId="keys">
-<TabItem value="insert" label="insert">
+<TabItem value="insert" label="insertKey">
 
-this command demonstrates a session key insertion using a key created with [subkey](https://docs.substrate.io/v3/tools/subkey).
+This command demonstrates a session key insertion using a key created with [subkey](https://docs.substrate.io/v3/tools/subkey).
 
-- generate an aura key with subkey
+:::note
+Starting with v3.2.1 you must provide all 3 of the following keys, earlier revisions needed only the Aura key
+:::
 
+- Generate an Aura key with subkey
   ```bash
   #!/bin/bash
-
   subkey generate \
     --scheme sr25519 \
     --network calamari \
     --output-type json \
     --words 12 \
     > ./aura.json
   ```
-
-- create an author_insertKey rpc payload
-
+- Generate a Nimbus key with subkey
+  ```bash
+  #!/bin/bash
+  subkey generate \
+    --scheme sr25519 \
+    --network calamari \
+    --output-type json \
+    --words 12 \
+    > ./nimbus.json
+  ```
+- Generate a VRF key with subkey
+  ```bash
+  #!/bin/bash
+  subkey generate \
+    --scheme sr25519 \
+    --network calamari \
+    --output-type json \
+    --words 12 \
+    > ./vrf.json
+  ```
+- Create an author_insertKey RPC payload
   ```bash
   #!/bin/bash
-
   echo '{
       "jsonrpc":"2.0",
       "id":1,
       "method":"author_insertKey",
       "params": [
         "aura",
-        "<mnemonic phrase>",
-        "<public key>"
+        "<mnemonic phrase1>",
+        "<public key1>",
+        "nimbus",
+        "<mnemonic phrase2>",
+        "<public key2>",
+        "vrf",
```

Comment:
should be `rand` if i'm not mistaken

Reply:
seems weird since if you check the session::set_keys screenshot, the param names are `nimbus` and `vrf`

It's possible this is using the key_id directly tho. Did you try executing this with your proposed names?

Reply:
Yeah, it worked with those. Otherwise I was getting an error telling me these can only be 4 characters.

Reply:
But I actually used the polkadot-js front end, maybe that's the difference